libp2p/Host.cpp

```diff
@@ -867,7 +868,10 @@ bytes Host::saveNetwork() const
             continue;
 
         // Only save peers which have connected within 2 days, with properly-advertised port and public IP address
-        if (chrono::system_clock::now() - p.m_lastConnected < chrono::seconds(3600 * 48) && !!p.endpoint && p.id != id() && (p.peerType == PeerType::Required || p.endpoint.isAllowed()))
+        if (chrono::system_clock::now() - p.m_lastConnected < chrono::seconds(3600 * 48) &&
+            !!p.endpoint && p.id != id() &&
+            (p.peerType == PeerType::Required ||
+                isAllowedAddress(m_netConfig.allowLocalDiscovery, p.endpoint.address())))
```

Comment:
Make a helper method `Host::isAllowedAddress(address)` to make this more concise and not to pass `m_netConfig.allowLocalDiscovery` each time.
Similar for `NodeTable`, too